nixos/modules/services/networking/networkmanager.nix

```diff
@@ -400,6 +408,12 @@ in {
           networking.networkmanager.dns = "dnsmasq"
         '';
       }
+      { assertion = !config.networking.wireless.iwd.enable;
```

Comment:
Would it be better to combine this assertion with the first one, since they're both trying to do the same thing albeit for different packages.

Also, if I'm not wrong, using iwd as the backend still requires `iwd.service` to be started as well?

Reply:
@eadwu How can the service be started?

Reply:
Since enabling `networking.wireless.iwd.enable` would trigger the assertion, you can try just manually applying the configuration [1]. I don't know whether or not NetworkManager uses the DBus implementation though.

[1] https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/iwd.nix#L18

Reply:
I'm not actually 100% sure the assertion should be there in the first place. I'll see if I have time to dig deeper.